agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -71,6 +71,7 @@ pub(crate) struct ResultSetState<'a> {
     pub variables: BTreeMap<String, Schema>,
     pub result_set_schema: Schema,
     pub current_db: String,
+    pub current_collection: Option<String>,
```

Comment:
this is the only file worth reviewing. All the test updates are just adding this field as necessary